apps/src/studio/ThreeSliceAudio.js

```diff
@@ -0,0 +1,131 @@
+/** @file A three-part loopable audio effect (start-loop-end). */
+/* jshint
+ funcscope: true,
+ newcap: true,
+ nonew: true,
+ shadow: false,
+ unused: true,
+ eqeqeq: true,
+
+ maxlen: 90,
+ maxstatements: 200
+ */
+'use strict';
+
+var debugLogging = false;
+function debug(msg) {
+  if (debugLogging && console && console.info) {
+    console.info('Audio: ' + msg);
+  }
+}
+
+/** @enum {number} */
+var PlaybackState = {
+  NONE: 'none',
+  BEGIN: 'begin',
+  LOOP: 'loop',
+  END: 'end'
+};
+
+/**
+ * A loopable audio effect defined with three parts (start, middle, end) so that
+ * you can get a smooth start/finish effect.
+ *
+ * Assumes the audio clips in question have already been preloaded by another
+ * system.
+ *
+ * @param {AudioPlayer} audioPlayer
+ * @param {string} begin - Audio clip name for start of sound.
+ * @param {string} loop - Audio clip name for loopable part of sound.
+ * @param {string} end - Audio clip name for end of sound.
```

Comment:
Possible improvement: Ability to pass other playback options (which studioApp.playSound takes) here that would apply to all parts of the sound effect.